pkg/lang/ir/system.go

```diff
@@ -118,6 +118,9 @@ func (g *Graph) compileBase() llb.State {
 	} else {
 		base = g.compileCUDAPackages()
 	}
+	if g.uid == 0 {
+		g.uid = 1000
```

Comment:
I used to think it works. But I am not sure since the working dir mounted in the container is also owned by root user.

Could you please test if it works?

Reply:
Let me do more test on it.  I overlooked there is a more step mount files on the directory.